clients/go/admin/auth_interceptor_test.go

```diff
@@ -0,0 +1,205 @@
+// initialized through mockery) and starts a local server that uses it to respond to grpc requests.
+type authMetadataServer struct {
+	s           *httptest.Server
+	t           testing.TB
+	port        int
+	grpcServer  *grpc.Server
+	netListener net.Listener
+	impl        service2.AuthMetadataServiceServer
+	lck         *sync.RWMutex
+}
+
+func (s authMetadataServer) GetOAuth2Metadata(ctx context.Context, in *service2.OAuth2MetadataRequest) (*service2.OAuth2MetadataResponse, error) {
+	return s.impl.GetOAuth2Metadata(ctx, in)
+}
+
+func (s authMetadataServer) GetPublicClientConfig(ctx context.Context, in *service2.PublicClientAuthConfigRequest) (*service2.PublicClientAuthConfigResponse, error) {
+	return s.impl.GetPublicClientConfig(ctx, in)
+}
+
+func (s authMetadataServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var issuer string
+	switch r.URL.Path {
+	case "/.well-known/oauth-authorization-server":
+		w.Header().Set("Content-Type", "application/json")
+		_, err := io.WriteString(w, strings.ReplaceAll(`{
+				"issuer": "https://dev-14186422.okta.com",
+				"authorization_endpoint": "https://example.com/auth",
+				"token_endpoint": "https://example.com/token",
+				"jwks_uri": "https://example.com/keys",
+				"id_token_signing_alg_values_supported": ["RS256"]
+			}`, "ISSUER", issuer))
+		if !assert.NoError(s.t, err) {
+			s.t.FailNow()
+		}
+
+		return
+	}
+
+	http.NotFound(w, r)
+}
+
+func (s *authMetadataServer) Start(_ context.Context) error {
+	s.lck.Lock()
+	defer s.lck.Unlock()
+
+	/***** Set up the server serving channelz service. *****/
+	lis, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", s.port))
+	if err != nil {
+		return fmt.Errorf("failed to listen on port [%v]: %w", s.port, err)
+	}
+
+	grpcS := grpc.NewServer()
+	service2.RegisterAuthMetadataServiceServer(grpcS, s)
+	go func() {
+		err := grpcS.Serve(lis)
+		assert.NoError(s.t, err)
+	}()
+
+	s.grpcServer = grpcS
+	s.netListener = lis
+
+	s.s = httptest.NewServer(s)
+
+	return nil
+}
+
+func (s *authMetadataServer) Close() {
+	s.lck.RLock()
+	defer s.lck.RUnlock()
+
+	s.grpcServer.Stop()
+	s.s.Close()
+}
+
+func newAuthMetadataServer(t testing.TB, port int, impl service2.AuthMetadataServiceServer) *authMetadataServer {
+	return &authMetadataServer{
+		port: port,
+		t:    t,
+		impl: impl,
+		lck:  &sync.RWMutex{},
+	}
+}
+
+func Test_newAuthInterceptor(t *testing.T) {
+	t.Run("Other Error", func(t *testing.T) {
+		f := NewPerRPCCredentialsFuture()
+		interceptor := newAuthInterceptor(&Config{}, &mocks.TokenCache{}, f)
+		otherError := func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, opts ...grpc.CallOption) error {
+			return status.New(codes.Canceled, "").Err()
+		}
+
+		assert.Error(t, interceptor(context.Background(), "POST", nil, nil, nil, otherError))
+	})
+
+	t.Run("Unauthenticated first time, succeed", func(t *testing.T) {
```

Comment:
is it possible to add a test for the authenticated code path or is that too hairy for unit tests?